packages/web/src/store/token-dashboard/signMessage.ts

```diff
@@ -24,9 +29,9 @@ export function* signMessage(connection: WalletConnection) {
     case Chain.Sol: {
       const encodedMessage = new TextEncoder().encode(message)
       const signedResponse = yield* call(
-        connection.provider.signMessage,
-        encodedMessage,
-        'utf8'
+        solSign,
```

Comment:
as @dharit-tan called out `[provider, provider.signMessage]` works here too, but it doesn't correctly infer the type of `provider`

Reply:
Great catch